docs/sphinx/examples/cpp/basics/cutensornet_backends.cpp

```diff
@@ -0,0 +1,42 @@
+// Compile and run with:
+// ```
+// nvq++ cutensornet_backends.cpp -o dyn.x --target tensornet
+// mpirun -np <N> ./dyn.x
+// ```
+
+// This example is meant to demonstrate the `cuTensorNet`
+// multi-node/multi-GPU backend.
+
+#include <cudaq.h>
+
+// Define a quantum kernel with a runtime parameter
+struct ghz {
+  auto operator()(const int N) __qpu__ {
+
+    // Dynamic, vector-like `qreg`
+    cudaq::qreg q(N);
+    h(q[0]);
+    for (int i = 0; i < N - 1; i++) {
+      x<cudaq::ctrl>(q[i], q[i + 1]);
+    }
+    mz(q);
+  }
+};
+
+int main() {
+  // Initialize MPI
+  cudaq::mpi::initialize();
+  if (cudaq::mpi::rank() == 0)
+    printf("Number of MPI processes: %d\n", cudaq::mpi::num_ranks());
+  auto counts = cudaq::sample(100, ghz{}, 28);
+  if (cudaq::mpi::rank() == 0) {
+    counts.dump();
```

Comment:
@bettinaheim can confirm or deny, but I believe the `validate_container.sh` script will attempt to run this without MPI, so it might be worth updating this code to also support it being launched without MPI. (See `bernstein_vazarani.cpp` for an example.)

When I tried to run it without MPI, it crashes for me with the following error:
```
Number of MPI processes: 1
[Warning] Enabling cuTensorNet MPI without environment variable CUTENSORNET_COMM_LIB.
MPI parallelization inside cuTensorNet library may cause an error.
cuTensorNet error CUTENSORNET_STATUS_DISTRIBUTED_FAILURE in line 39
[d1b458e9d687:04497] *** Process received signal ***
[d1b458e9d687:04497] Signal: Aborted (6)
[d1b458e9d687:04497] Signal code:  (-6)
```

Reply:
Good point. I'll fix the example to not using `mpi` by default and adding comments on how to enable it.